src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1320,21 +1321,27 @@ synchronized public void onChangePredicate(Chat chat, String predicateName, Stri
     for (Session s : sessions.values()) {
       if (s.chat == chat) {
         // found session saving predicates
-        invoke("publishChangePredicate", s, chat, predicateName, result);
+        invoke("publishPredicate", s, chat, predicateName, result);
         s.savePredicates();
         return;
       }
     }
     error("could not find session to save predicates");
   }
 
-  public PredicateEvent publishChangePredicate(Session session, Chat chat, String name, String value) {
+  public PredicateEvent publishPredicate(Session session, Chat chat, String name, String value) {

```

Comment:
Why the rename? Would this break any services or scripts subscribing to it?

Also would appreciate a Java doc on it, having difficulty figuring out what the parameters do

Reply:
Checked nothing depends on it, and its new functionality I've accounted for in the UI
Added docs

Reply:
Not sure changing the name has enough of a benefit, considering it's publishing a change of a predicate. User scripts may rely on it too (unlikely, but the possibility is there)

Reply:
good point, but its relatively new .. and in the discussion with @kwatters we agreed on publishTopic .. so it would be following the same convention